sys/prebuilt-bindings/geos_3.6.rs

```diff
@@ -1,1285 +1,2048 @@
-    pub fn GEOSPolygonize(geoms: *const *const GEOSGeometry, ngeoms: c_uint) -> *mut GEOSGeometry;
-    pub fn GEOSPolygonize_full(
-        input: *const GEOSGeometry,
-        cuts: *mut *mut GEOSGeometry,
-        dangles: *mut *mut GEOSGeometry,
-        invalidRings: *mut *mut GEOSGeometry,
-    ) -> *mut GEOSGeometry;
-    pub fn GEOSPolygonizer_getCutEdges(
-        geoms: *const *const GEOSGeometry,
-        ngeoms: c_uint,
-    ) -> *mut GEOSGeometry;
-    pub fn GEOSProject(g: *const GEOSGeometry, p: *const GEOSGeometry) -> c_double;
-    pub fn GEOSRelatePattern(
-        g1: *const GEOSGeometry,
-        g2: *const GEOSGeometry,
-        pat: *const c_char,
-    ) -> c_char;
-    pub fn GEOSRelate(g1: *const GEOSGeometry, g2: *const GEOSGeometry) -> *mut c_char;
-    pub fn GEOSRelatePatternMatch(mat: *const c_char, pat: *const c_char) -> c_char;
-    pub fn GEOSRelateBoundaryNodeRule(
-        g1: *const GEOSGeometry,
-        g2: *const GEOSGeometry,
-        bnr: c_int,
-    ) -> *mut c_char;
-    pub fn GEOSSharedPaths(g1: *const GEOSGeometry, g2: *const GEOSGeometry) -> *mut GEOSGeometry;
-    pub fn GEOSWKBReader_create() -> *mut GEOSWKBReader;
-    pub fn GEOSWKBReader_destroy(reader: *mut GEOSWKBReader);
-    pub fn GEOSWKBReader_read(
-        reader: *mut GEOSWKBReader,
-        wkb: *const c_uchar,
-        size: size_t,
-    ) -> *mut GEOSGeometry;
-    pub fn GEOSWKBReader_readHEX(
-        reader: *mut GEOSWKBReader,
-        hex: *const c_uchar,
-        size: size_t,
-    ) -> *mut GEOSGeometry;
-    pub fn GEOSWKBWriter_create() -> *mut GEOSWKBWriter;
-    pub fn GEOSWKBWriter_destroy(reader: *mut GEOSWKBWriter);
-    pub fn GEOSWKBWriter_write(
-        writer: *mut GEOSWKBWriter,
-        g: *const GEOSGeometry,
-        size: *mut size_t,
-    ) -> *mut c_uchar;
-    pub fn GEOSWKBWriter_writeHEX(
-        writer: *mut GEOSWKBWriter,
-        g: *const GEOSGeometry,
-        size: *mut size_t,
-    ) -> *mut c_uchar;
-    pub fn GEOSWKBWriter_getOutputDimension(writer: *const GEOSWKBWriter) -> c_int;
-    pub fn GEOSWKBWriter_setOutputDimension(writer: *mut GEOSWKBWriter, newDimension: c_int);
-    pub fn GEOSWKBWriter_getByteOrder(writer: *const GEOSWKBWriter) -> c_int;
-    pub fn GEOSWKBWriter_setByteOrder(writer: *mut GEOSWKBWriter, byteOrder: c_int);
-    pub fn GEOSWKBWriter_getIncludeSRID(writer: *const GEOSWKBWriter) -> c_char;
-    pub fn GEOSWKBWriter_setIncludeSRID(writer: *mut GEOSWKBWriter, writeSRID: c_char);
-    pub fn GEOSisValidDetail(
-        g: *const GEOSGeometry,
-        flags: c_int,
-        reason: *mut *mut c_char,
-        location: *mut *mut GEOSGeometry,
-    ) -> c_char;
-    pub fn GEOS_interruptCancel();
     pub fn GEOS_interruptRequest();
-    pub fn GEOS_interruptRegisterCallback(
-        cb: *mut GEOSInterruptCallback,
-    ) -> *mut GEOSInterruptCallback;
-
-    /* Thread safe calls */
+}
+extern "C" {
```

Comment:
It's really bad that it's generating a lot of `extern "C"` blocks. Could you merge them please?

Reply:
This seems to be an artifact of how `bindgen` currently works and has an open [feature request](https://github.com/rust-lang/rust-bindgen/issues/564) to merge them.  I didn't see any options at present that would allow us to merge these, and since it is automatically generated code, I think this is the output we get.  Future versions of bindgen may address this.

Other than readability (which would indeed be nicer if merged), is there a performance impact?

Reply:
I don't think so. Not great but for now let's keep as is.